service/apis/meta_service.go

```diff
@@ -105,3 +107,70 @@ func (s MetaService) GetMetadataFields(name string) ([]MetadataField, error) {
 	}
 	return nil, nil
 }
+
+// CheckKeys checks if keys exist
+func CheckKeys(keys []string, data map[string]interface{}) error {
+	for k := range keys {
+		if _, ok := data[keys[k]]; !ok {
+			return fmt.Errorf("%q doesn't exist", keys[k])
+		}
+	}
+	return nil
+}
+
+//SetDefaultKey sets default values for the keys if any isn't set
+func SetDefaultKey(dkMap map[string]interface{}, data map[string]interface{}) error {
+	for key := range dkMap {
+		if _, ok := data[key]; !ok {
+			data[key] = dkMap[key]
+			log.Infof("%v doesn't exist. set to default %#v", key, dkMap[key])
+		}
+	}
+	return nil
+}
+
+// CreateMetadata save new metadata to the storage
+func (s MetaService) CreateMetadata(data map[string]interface{}) (map[string]string, error) {
+	var rkeys = []string{"name", "fields", "objtype"}
+	err := CheckKeys(rkeys, data)
+	if err != nil {
+		return nil, err
+	}
+	fMap, ok := data["fields"].([]interface{})
+	if ok {
+		log.Debugf("fMap is %#v", fMap)
+	}
+	const cardinality = "One"
```

Comment:
should the code run to this line if it is not ok?

Reply:
Definitely my oversight